wallet/src/Pos/Wallet/Web/Tracking/Sync.hs

```diff
@@ -480,11 +480,8 @@ rollbackModifierFromWallet wid newTip CAccModifier{..} = do
     removeFromHead :: [TxHistoryEntry] -> [TxHistoryEntry] -> [TxHistoryEntry]
     removeFromHead [] ths = ths
     removeFromHead _ [] = []
-    removeFromHead (dTh : dThes) (th : thes) =
-        if _thTxId dTh == _thTxId th
-        then removeFromHead dThes thes
-        else error "rollbackModifierFromWallet: removeFromHead: \
-                   \rollbacked tx ID is not present in history cache!"
+    removeFromHead (dTh : dThes) thes =
+        removeFromHead dThes $! deleteBy ((==) `on` _thTxId) dTh thes
```

Comment:
What is the consequence here? Sure we don't report and error, but presumably it is a violation of a data structure invariant. We will still get a wrong history list.

Reply:
@dcoutts Sure thing, but having incorrect order of transactions in some part of history is way better than having basically invalid wallet db with broken balances, history and no ability to create a valid transaction.
I think that the best solution here would be to log a warning when `dTh` is not equal to first element of `thes`, but proceed with `deleteBy` anyway - that way we'll see when incorrect history happens in logs, but won't have dire consequences.

Reply:
IMHO this change shouldn't be included to release branch until we will be sure it's fixed

Reply:
@pva701 well, if we understand correctly this is in fact a crucial fix to stop the problem reoccurring. Yes we must test it.

Reply:
I tested it via making many transactions for several slots, then initiaiting rollback artificially. It worked even when I permuted transactions passed to `removeFromHead`